### PR TITLE
Adds an `isLoading` prop to `DataTable`

### LINKED ISSUE #557: `DataTable` needs an `isLoading` props

### PR DESCRIPTION
**Pull request by mute**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~[ ] One core team UX approval~~

Fixes #557

### CODE REVIEW

File: src/components/DataTable/DataTable.jsx [hunk 1]
```diff
@@ -237,17 +240,12 @@ const DataTable = createClass({
 		const emptyMessageTitleProp = _.get(getFirst(this.props, DataTable.EmptyMessageTitle), 'props', {children: 'You have no items.'});
 
 		return (
-			<OverlayWrapper
-				hasOverlay={false}
-				isVisible={!data || data.length === 0}
+			<DataTableWrapper
+				isLoading={isLoading}
+				isEmpty={!data || data.length === 0}
```
Comment (sodiumjoe) [inline src/components/DataTable/DataTable.jsx]: i think `_.isEmpty` would do here

File: src/components/DataTableWrapper/DataTableWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["utility"], "madeFrom": ["HatchPattern", "LoadingIndicator", "OverlayWrapper"]}
+ *
+ * A wrapper for the DataTable which can display either a `LoadingIndicator` or `OverlayWrapper`.
+ *
+ */
+const DataTableWrapper = createClass({
```
Comment (sodiumjoe) [inline src/components/DataTableWrapper/DataTableWrapper.jsx]: this is really clean, i like it

Comment (sodiumjoe) [inline src/components/DataTableWrapper/DataTableWrapper.jsx]: Seems like there's nothing table specific here, could we reuse this for charts?

Comment (mute) [inline src/components/DataTableWrapper/DataTableWrapper.jsx]: I think that's a great idea and we totally could.

**Pull request merged**